src/core/document.js

```diff
@@ -714,6 +697,61 @@ class PDFDocument {
     return shadow(this, "numPages", num);
   }
 
+  /**
+   * @private
+   */
+  _hasOnlyDocumentSignatures(fields) {
+    return fields.every(field => {
+      field = this.xref.fetchIfRef(field);
+      if (field.has("Kids")) {
+        return this._hasOnlyDocumentSignatures(field.get("Kids"));
+      }
+      const fieldType = field.get("FT");
+      const isSignature = isName(fieldType) && fieldType.name === "Sig";
+      const rectangle = field.get("Rect");
+      const isInvisible =
+        Array.isArray(rectangle) && rectangle.every(value => value === 0);
+      return isSignature && isInvisible;
+    });
+  }
+
+  get formInfo() {
+    const formInfo = { hasAcroForm: false, hasXfa: false };
+    try {
+      const acroForm = this.catalog.acroForm;
+      if (acroForm) {
```

Comment:
Also, with the error handling added to `Catalog.acroForm` you can swap these lines around a little bit:
```suggestion
    const acroForm = this.catalog.acroForm;
    if (acroForm) {
      try {
```